### DISCUSSION TITLE
Need Assistance UNet from MMSeg for Regression Within MMSelfSup

### DISCUSSION THREAD

**Discussion by Shaviv-Hoffman-Lowitz** [open-mmlab/mmselfsup#598] [How-to]
I am trying to use a UNet for a regression task (predicting continuous values) within MMSelfSup, and since there was no UNet defined in the MMSelfSup repository but there was one defined in MMSeg, I have been working with that UNet. However, since it seems that all of the example 'heads' and 'algorithms' in MMSelfSup are for classification tasks, I was wondering if there are any examples for regression tasks that I might be missing? If not, is there an easy way to adapt the BaseDecodeHead from MMSeg (https://github.com/open-mmlab/mmsegmentation/blob/441be4e435127868a0c72a4e0e6b87662a4c415b/mmseg/models/decode_heads/decode_head.py) to a regression setting, instead of a classification setting, so that I can use a UNet for regression in the MMSelfSup repository?

Thank you so very much for any input, I sincerely appreciate it!

**Comment by PranayShirodkar** [comment]
Hello! Shaviv and I are working on the same project, I just wanted to add some more information:

We are trying to do climate forecasting. The model input is a 2D matrix of temperature values at time=0 and the output is a 2D matrix of temperature values at time=1. We use temperature as a continuous variable so there is no classification. We want to use UNet for this task, and we have managed to pre-train the UNet model using SimCLR (we think the pretraining worked). Now for the downstream task/fine-tuning, we are trying to figure out which 'head' and 'algorithm' to use, or if we need to create our own. I hope that adds some background to the question.

We would really appreciate any advice, thank you!!

**Comment by fangyixiao18** [comment]
Hello, for regression task, I think MAE also predicting continuous values, the link is here, https://github.com/open-mmlab/mmselfsup/blob/1.x/mmselfsup/models/heads/mae_head.py

Hope it will be helpful.

**Reply by PranayShirodkar** [reply]
Thank you very much for your response! We will give that a try
